src/libexpr/eval-inline.hh

```diff
@@ -24,6 +24,81 @@ LocalNoInlineNoReturn(void throwTypeError(const Pos & pos, const char * s, const
 }
 
 
+/* Note: Various places expect the allocated memory to be zeroed. */
+[[gnu::always_inline]]
+inline void * allocBytes(size_t n)
+{
+    void * p;
+#if HAVE_BOEHMGC
+    p = GC_MALLOC(n);
+#else
+    p = calloc(n, 1);
+#endif
+    if (!p) throw std::bad_alloc();
+    return p;
+}
+
+
+[[gnu::always_inline]]
```

Comment:
Is `always_inline` a good idea? Maybe this is best left to the compiler.

Reply:
in general no, but in this case it does seem to be. it does save the entire stack frame setup and teardown, which can be a significant part of allocation cost when taking the hot path using the cache. if the cache weren't there it'd definitely be better left to the compiler.